src/Symfony/Bundle/TwigBundle/TwigEngine.php

```diff
@@ -74,10 +74,14 @@ public function render($name, array $parameters = array())
             if ($name instanceof TemplateReference) {
                 try {
                     // try to get the real name of the template where the error occurred
-                    $name = $e->getTemplateName();
-                    $path = (string) $this->locator->locate($this->parser->parse($name));
+                    if (method_exists($e, 'getSourceContext')) {
+                        $templateName = $e->getSourceContext()->getName();
```

Comment:
@nicolas-grekas @attila901 maybe ```$path``` could be also get from the context with ```$e->getSourceContext()->getPath()``` in the else ```$path``` could be set the old way.